src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,126 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double GROWTH_FACTOR = 1.5;
+
+    private T[] elements;
+    private int size;
+
+    public ArrayList() {
+        this.elements = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        ensureCapacity();
+        elements[size++] = value;
```

Comment:
The 'ensureCapacity' method name is not completely informative. It should reflect the fact that it only ensures capacity if the array is full. Consider renaming it to something like 'ensureCapacityIfFull' or 'growIfNeeded'.